src/language/interpreter_new.rs

```diff
@@ -1019,316 +1019,317 @@ impl Sqrt for i64 {
     }
 }
 
+extern crate test;
+
 #[cfg(test)]
 mod tests {
+
     use super::*;
     use approx::AbsDiffEq;
     use ndarray::array;
     use std::str::FromStr;
+    use test::Bencher;
+
+    /// Creates an benchmark test for the interpreter
+    /// The test does the following:
+    ///  1. Parses $glenside_str as glenside expression
+    ///  2. Creates a new Environment from the vector of (key, value) pairs
+    ///  3. Calls the interpreter with the glenside expression and env,
+    ///         and passes the value to check_correct
+    /// $test_name: the name of the created benchmark test
+    /// $glenside_str: A string containing the Glenside program
+    /// $env: A vector of 2-tuples of key, value pairs to put into the environment
+    /// $check_correct: A closure with arguments (value) that checks for correctness
+    macro_rules! benchmark_test {
+        ($test_name:ident, $glenside_str: expr, $env: expr, $check_correct: expr) => {
+            #[bench]
+            fn $test_name(b: &mut Bencher) {
+                let test = test_function!($glenside_str, $env, $check_correct);
+                b.iter(|| test());
```

Comment:
I just realized that we're currently benchmarking all of the setup and checking (i.e. creating the environment, parsing the Glenside string, and checking that the result is correct). Ideally we wouldn't do this, but getting around that might actually be a pain, as I'm looking into it. There are a few possible ways forward, but for now, I think what you have here will work fine for our purposes!